TipCatDotNet.Api/Services/Analitics/AccountResumeService.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using CSharpFunctionalExtensions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using TipCatDotNet.Api.Data;
+using TipCatDotNet.Api.Data.Analitics;
+using TipCatDotNet.Api.Data.Models.Payment;
+using TipCatDotNet.Api.Infrastructure.Logging;
+
+namespace TipCatDotNet.Api.Services.Analitics;
+
+public class AccountResumeService : IAccountResumeService
+{
+    public AccountResumeService(AetherDbContext context, ILoggerFactory loggerFactory)
+    {
+        _context = context;
+        _logger = loggerFactory.CreateLogger<AccountResumeService>();
+    }
+
+
+    public async Task<Result> CreateOrUpdate(Transaction transaction, CancellationToken cancellationToken = default)
+    {
+        var accountId = await _context.Facilities
+        .Where(m => m.Id == transaction.FacilityId)
+        .Select(m => m.AccountId)
+        .SingleOrDefaultAsync();
```

Comment:
You don't check `accountId` for defaults below, so it could be simply `SingleAsync()`